onnxruntime/python/tools/transformers/models/llama/llama_parity.py

```diff
@@ -0,0 +1,163 @@
+    }
+    if args.use_past_kv:
+        inputs["input_ids"] = inputs["input_ids"][:, -1:]
+
+        inputs.update(
+            {
+                "past_key_values": [
+                    (
+                        torch.rand(batch_size, num_heads, seq_len - 1, head_size),
+                        torch.rand(batch_size, num_heads, seq_len - 1, head_size),
+                    )
+                    for _ in range(config.num_hidden_layers)
+                ]
+            }
+        )
+
+    pt_outputs = pt_model(**inputs).logits.detach().cpu().numpy()
+
+    # Run inference with ORT
+    ort_model = create_onnxruntime_session(
+        args.onnxruntime,  # onnx_model_path
+        args.execution_provider != "cpu",  # use_gpu
+        provider=args.execution_provider,
+        verbose=args.verbose,
+    )
+
+    for k, v in inputs.items():
+        if k == "past_key_values":
+            continue
+        else:
+            inputs[k] = v.detach().cpu().numpy()
+
+    if "past_key_values" in inputs:
+        np_dtype = np.float16 if ort_model.get_outputs()[0].type == "tensor(float16)" else np.float32
+        for i, (past_k, past_v) in enumerate(inputs["past_key_values"]):
+            inputs[f"past_key_values.{i}.key"] = past_k.detach().cpu().numpy().astype(np_dtype)
+            inputs[f"past_key_values.{i}.value"] = past_v.detach().cpu().numpy().astype(np_dtype)
+        del inputs["past_key_values"]
+
+    ort_outputs = ort_model.run(None, inputs)[0][0]
+
+    # Compare PyTorch and ONNX Runtime accuracy
+    tol = 1e-3 if "fp32" in args.onnxruntime else 1e-2 if "fp16" in args.onnxruntime else 1e2
+    parity = np.allclose(pt_outputs, ort_outputs, rtol=tol, atol=tol)
+    logger.warning(f"Are PyTorch and ONNX Runtime results close? {parity}")
+    if not parity:
+        logger.warning(f"Max diff: {np.max(pt_outputs - ort_outputs)}")
+
+
+def get_args(argv: List[str]):
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "-m",
+        "--model_name",
+        required=True,
+        help="Model name in Hugging Face",
+    )
+
+    parser.add_argument(
+        "-p",
+        "--pytorch",
+        required=False,
+        default=os.path.join("."),
+        help="Directory to PyTorch model and associated files if saved on disk",
+    )
+
+    parser.add_argument(
+        "-o",
+        "--onnxruntime",
+        required=True,
+        default=os.path.join("."),
+        help="Directory to ONNX Runtime model and associated files saved on disk",
+    )
+
+    parser.add_argument(
+        "-ep",
+        "--execution_provider",
+        required=False,
+        default="cpu",
+        choices=["cpu", "cuda", "rocm"],
+        help="Execution provider to verify parity with",
+    )
+
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action="store_true",
+        help="Print verbose logs",
+    )
+    parser.set_defaults(verbose=False)
+
+    parser.add_argument(
+        "-kv",
```

Comment:
`-kv` is not a good abbreviation. May use `-p` here, then use `-t` for torch directory to avoid confliction.

Reply:
Fixed